File: Frontend/src/types/Agente.ts
export type RolAgente = 'lider' | 'arquitecto' | 'backend' | 'frontend' | 'qa' | 'dba'
export type EstadoAgente = 'activo' | 'pensando' | 'inactivo' | 'error'

export interface Agente {
  id: string
  proyectoId: string
  rol: RolAgente
  nombre: string
  especialidad?: string
  estado: EstadoAgente
  fechaIncorporacion: string
}

export interface DatosNuevoAgente {
  rol: RolAgente
  nombre: string
  especialidad?: string
}

export const COLOR_POR_ROL: Record<RolAgente, string> = {
  lider: '#00d4ff',
  arquitecto: '#7c3aed',
  backend: '#ff8c42',
  frontend: '#ec4899',
  qa: '#00ff88',
  dba: '#ffd700',
}

export const NOMBRE_ROL: Record<RolAgente, string> = {
  lider: 'Líder',
  arquitecto: 'Arquitecto',
  backend: 'Backend Dev',
  frontend: 'Frontend Dev',
  qa: 'QA Engineer',
  dba: 'DBA',
}

File: Backend/models/agente.py
from enum import Enum
from pydantic import BaseModel
from datetime import datetime


class RolAgente(str, Enum):
    LIDER = "lider"
    ARQUITECTO = "arquitecto"
    BACKEND = "backend"
    FRONTEND = "frontend"
    QA = "qa"
    DBA = "dba"


class EstadoAgente(str, Enum):
    ACTIVO = "activo"
    PENSANDO = "pensando"
    INACTIVO = "inactivo"
    ERROR = "error"


class Agente(BaseModel):
    id: str
    proyecto_id: str
    rol: RolAgente
    nombre: str
    especialidad: str | None = None
    estado: EstadoAgente = EstadoAgente.INACTIVO
    fecha_incorporacion: datetime

    class Config:
        from_attributes = True


class DatosNuevoAgente(BaseModel):
    rol: RolAgente
    nombre: str
    especialidad: str | None = None


class RespuestaAgente(BaseModel):
    agente: Agente
    mensaje: str | None = None
